metricbeat/module/postgresql/dbcache.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package postgresql
+
+import (
+	"context"
+	"database/sql"
+	"sync"
+)
+
+type cacheKey struct {
+	driver string
+	uri    string
+}
+
+type dbCacheType struct {
+	dbs  map[cacheKey]*sql.DB
+	lock sync.Mutex
```

Comment:
Nit: What about making this just `sync.Mutex` to compose dbCacheType with it? Like this below it would be just `cache.Lock()`.